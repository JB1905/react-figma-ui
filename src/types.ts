import { colorNames, iconNames, sizes, tints, weights } from './constants';

export type ColorName = (typeof colorNames)[number];
export type IconName = (typeof iconNames)[number];
export type Size = (typeof sizes)[number];
export type Tint = (typeof tints)[number];
export type Weight = (typeof weights)[number];
