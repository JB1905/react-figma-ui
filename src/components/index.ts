export * from './Button';
export * from './Checkbox';
export * from './Disclosure';
export * from './Icon';
export * from './IconButton';
export * from './Input';
export * from './Label';
export * from './Onboarding';
export * from './Radio';
export * from './SectionTitle';
export * from './SelectMenu';
export * from './Switch';
export * from './Textarea';
export * from './Type';
