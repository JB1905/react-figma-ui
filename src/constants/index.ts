export * from './colorNames';
export * from './iconNames';
export * from './sizes';
export * from './tints';
export * from './weights';
