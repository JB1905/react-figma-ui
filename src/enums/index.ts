export { Size } from './Size';
export { Tint } from './Tint';
export { Weight } from './Weight';
