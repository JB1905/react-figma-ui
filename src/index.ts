export { Button } from './components/Button';
export { Checkbox } from './components/Checkbox';
export { Disclosure } from './components/Disclosure';
export { Icon } from './components/Icon';
export { IconButton } from './components/IconButton';
export { Input } from './components/Input';
export { Label } from './components/Label';
export { Onboarding } from './components/Onboarding';
export { Radio } from './components/Radio';
export { SectionTitle } from './components/SectionTitle';
export { SelectMenu } from './components/SelectMenu';
export { Switch } from './components/Switch';
export { Textarea } from './components/Textarea';
export { Type } from './components/Type';
