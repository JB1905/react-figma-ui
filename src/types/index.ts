import { colorNames } from '../constants/colorNames';
import { iconNames } from '../constants/iconNames';
import { sizes } from '../constants/sizes';
import { tints } from '../constants/tints';
import { weights } from '../constants/weights';

export type ColorName = typeof colorNames;
export type IconName = typeof iconNames;
export type Size = typeof sizes;
export type Tint = typeof tints;
export type Weight = typeof weights;
