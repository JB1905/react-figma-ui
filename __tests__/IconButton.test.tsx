import React from 'react';
import { render } from '@testing-library/react';

import { IconButton } from '../src';

describe('IconButton', () => {
  it('should render Icon Button with a blend icon', () => {
    const { container } = render(
      <IconButton iconProps={{ iconName: 'blend' }} />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render Icon Button with selected option', () => {
    const { container } = render(
      <IconButton iconProps={{ iconName: 'blend' }} selected />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render IconButton with custom className', () => {
    const { container } = render(
      <IconButton
        iconProps={{ iconName: 'blend' }}
        className="custom-class-name"
      />
    );

    expect(container).toMatchSnapshot();
  });

  // TODO disabled
  // TODO onClick
});
