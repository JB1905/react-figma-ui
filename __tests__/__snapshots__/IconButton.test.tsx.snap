// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton should render Icon Button with a blend icon 1`] = `
<div>
  <button
    class="icon-button"
  >
    <div
      class="icon icon--blend"
    />
  </button>
</div>
`;

exports[`IconButton should render Icon Button with selected option 1`] = `
<div>
  <button
    class="icon-button icon-button--selected"
  >
    <div
      class="icon icon--blend"
    />
  </button>
</div>
`;

exports[`IconButton should render IconButton with custom className 1`] = `
<div>
  <button
    class="icon-button custom-class-name"
  >
    <div
      class="icon icon--blend"
    />
  </button>
</div>
`;
